File: frontend/src/app/models/domain/school-class.dto.ts

export class SchoolClassDTO {

    // FIELDS

    private id: number;
    private name: string;
    private start: Date;
    private end: Date;

    // CONSTRUCTOR

    constructor() {}

    // GETTERS / SETTERS

    public getId(): number {
        return this.id;
    }

    public setId(id: number) {
        this.id = id;
    }

    public getName(): string {
        return this.name;
    }

    public setName(name: string) {
        this.name = name;
    }

    public getStart(): Date {
        return this.start;
    }

    public setStart(start: Date) {
        this.start = start;
    }

    public getEnd(): Date {
        return this.end;
    }

    public setEnd(end: Date) {
        this.end = end;
    }
}

File: backend/src/main/java/com/williamdsw/semsys/domain/dto/SchoolClassDTO.java
package com.williamdsw.semsys.domain.dto;

import java.io.Serializable;
import java.util.Date;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.williamdsw.semsys.domain.SchoolClass;

public class SchoolClassDTO implements Serializable {
	// FIELDS

	private static final long serialVersionUID = 1L;
	private Integer id;
	private String name;

	@JsonFormat(pattern = "yyyy-MM-dd")
	private Date start;

	@JsonFormat(pattern = "yyyy-MM-dd")
	private Date end;

	// CONSTRUCTORS

	public SchoolClassDTO() {
	}

	public SchoolClassDTO(SchoolClass schoolClass) {
		super();
		this.id = schoolClass.getId();
		this.name = schoolClass.getName();
		this.start = schoolClass.getStart();
		this.end = schoolClass.getEnd();
	}

	// GETTERS / SETTERS

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public Date getStart() {
		return start;
	}

	public void setStart(Date start) {
		this.start = start;
	}

	public Date getEnd() {
		return end;
	}

	public void setEnd(Date end) {
		this.end = end;
	}
}